CMake/HighFiveTargetDeps.cmake

```diff
@@ -0,0 +1,58 @@
+# Link against target system libs
+# -------------------------------
+
+# therefore making it possible to have new dependencies each build
+if(NOT TARGET highfive_deps)
+  add_library(highfive_deps INTERFACE)
+else()
+# Reset if imported
+set_target_properties(highfive_deps PROPERTIES
+  INTERFACE_COMPILE_DEFINITIONS ""
+  INTERFACE_COMPILE_OPTIONS ""
+  INTERFACE_INCLUDE_DIRECTORIES ""
+  INTERFACE_LINK_LIBRARIES ""
+  INTERFACE_LINK_OPTIONS ""
+  INTERFACE_SYSTEM_INCLUDE_DIRECTORIES ""
+)
+endif()
+
+
+# HDF5
+if(NOT DEFINED HDF5_C_LIBRARIES)
+  set(HDF5_PREFER_PARALLEL ${HIGHFIVE_PARALLEL_HDF5})
+  find_package(HDF5 REQUIRED)
+endif()
+
+if(HIGHFIVE_PARALLEL_HDF5 AND NOT HDF5_IS_PARALLEL)
+  message(WARNING "Parallel HDF5 requested but libhdf5 doesnt support it")
+endif()
+
+target_include_directories(highfive_deps SYSTEM INTERFACE ${HDF5_INCLUDE_DIRS})
+target_link_libraries(highfive_deps INTERFACE ${HDF5_C_LIBRARIES})
+target_compile_definitions(highfive_deps INTERFACE ${HDF5_DEFINITIONS})
+
+# Boost
+if(USE_BOOST)
```

Comment:
It would be good the use `HIGHFIVE_USE_BOOST` to avoid any name conflicts with other libraries

Reply:
I agree. It's a breaking change though. Let's do that in the next main release

Reply:
You could ease the transition by externally allowing both and internally using `HIGHFIVE_USE_BOOST`. One could even add a deprecation warning.